samples/99_zil_interpreter/app/main.rb

```diff
@@ -7,39 +7,108 @@
 
 def tick(args)
   setup(args) if args.tick_count.zero?
-  # TODO:
-  # - Render history
-  # - Collect input
-  # - Call send_input with input when pressing Enter
+  # Render history
+  render_state(args)
+  
+  # Collect input
+  handle_input(args)
+  
+  # Call send_input when pressing Enter
+  if args.inputs.keyboard.key_down.enter
+    send_input(args, args.state.input)
+  end
+  
   $gtk.request_quit unless $interpreter.alive?
 end
 
 def setup(args)
+  args.state.text_color = [240, 255, 255]
+  args.state.bg_color = [0, 0, 0]
+  args.state.input = ''
+  args.state.text_history = []
+  args.state.current_line = 0
```

Comment:
```suggestion
  args.state.history_offset = 0
```

calling it `history_offset` might make the meaning of the number more clearer